File: Python3/Medium/WrongAnswer/string-compression_1.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        
        i = 0
        while i < len(chars):
            c = chars[i]
            j = i + 1
            while j < len(chars) and chars[j] == c:
                j += 1
            if j - i > 1:
                for k, d in enumerate(str(j - i)):
                    chars[i + k + 1] = d
                del chars[i + k + 2:j]
            i = j
        return len(chars)

File: Rust/Medium/RuntimeError/string-compression_1.rs
impl Solution {
    pub fn compress(chars: &mut Vec<char>) -> i32 {
        

        let mut i = 0;
        let mut j = 0;
        while i < chars.len() {
            let mut count = 1;
            while j + 1 < chars.len() && chars[j] == chars[j + 1] {
                j += 1;
                count += 1;
            }
            chars[i] = chars[j];
            i += 1;
            if count > 1 {
                for c in count.to_string().chars() {
                    chars[i] = c;
                    i += 1;
                }
            }
            j += 1;
        }
        i as i32
    }
}